### ISSUE TITLE
[Question] Polyline.addPoint versus setPoints API

### ISSUE THREAD

**Issue by pasniak** [osmdroid/osmdroid#1101]
TLDR: Do I have to call Polyline setPoints with an empty array before any addPoint? 

I was testing my app in flight and got an NPE when adding points to a map track (Polyline). I only have a stack. In a postmortem analysis I realized there is some asymmetry between Polyline's setPoints and addPoint API:

Polyline.addPoint(p) assumes mOriginalPoints/mOutline are _not_ null while Polyline.setPoints does         mOriginalPoints = new ArrayList(points.size())

Why so? I would expect to just create a Polyline() and call addPoint... 

Shall I call setPoints with an empty array before any addPoint?  

BTW: I create Polyline without a map, in a background service.... so I call Polyline's constructor without the MapView....

https://github.com/osmdroid/osmdroid/blob/fa144b43ba00229db99bc8244f4c3182bfed2aaa/osmdroid-android/src/main/java/org/osmdroid/views/overlay/Polyline.java#L48

**Comment by spyhunter99** [comment]
hmm not too sure @MKergall recently rewrote that bit.

I think setPoints clears everything and uses the supplied points. Most of the time, i use this.

**Comment by pasniak** [comment]
setPoints works for "definitive" polylines, stuff that does not changes....  I have 1000-point long track to draw as polyline, I'd rather add points to it rather than recreate it every second or so...

**Comment by MKergall** [comment]
addPoint should work without any setPoints before, as mOriginalPoints/mOutline are initialized at declaration. Just looking at the code, I've not found the issue. 
What is your stack trace telling?

**Comment by haggaishG** [comment]
I got the same problem. 
setPoints() internally calls mOutline.clearPath() while mOutline is null
This happens every second time I run the application, which is odd. 

From what I can see, mOutline is initiated at declaration, and only get null in onDetach()

Now how come I get to access a detached object?
